controlplane/api/v1beta1/rke2controlplane_types.go

```diff
@@ -77,9 +77,9 @@ type RKE2ControlPlaneSpec struct {
 	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
 
 	// RegistrationMethod is the method to use for registering nodes into the RKE2 cluster.
-	// +kubebuilder:validation:Enum=internal-first;internal-only-ips;external-only-ips;address;control-plane-endpoint
+	// +kubebuilder:validation:Enum=internal-first;internal-only-ips;external-only-ips;address;control-plane-endpoint;""
 	// +optional
-	RegistrationMethod RegistrationMethod `json:"registrationMethod"`
+	RegistrationMethod RegistrationMethod `json:"registrationMethod,omitempty"`
```

Comment:
By PR description, the first thing that came to my mind was seeing something like: `+kubebuilder:default` for the `control-plane-endpoint` registration method. Is that a wrong assumption?

Reply:
That would unfortunately break the ADR progression on this topic: https://github.com/rancher/cluster-api-provider-rke2/blob/main/docs/adr/0002-deprecate-kubebuilder-defaults.md#context - having a default value causes the field to be incompatible with gitops, since this field will cause a state drift by being defaulted by the API server